test/slq.jl

```diff
@@ -0,0 +1,157 @@
+using Test
+using LinearAlgebra
+using TraceEstimation
+using SparseArrays
+
+
+function error(computed::Float64, actual::Float64)
+    if (abs(computed - actual)/actual) < 100
+        return true
+    else
+        return false
+    end
+end
+
+function signCheck(eMax::Float64, eMin::Float64)
+    if sign(eMax) == sign(eMin)
+        return true
+    else
+        retuen false
+    end
+end
+
+eps = rand(1)
+neta = rand(1)
+f(x) = exp(x)
+g(x) = x^2
+
+@testset "Testing Stochastic Lanczos Quadrature" begin
+    @testset "Testing for Dense Matrices" begin
+        @testset "For Symmetric Matrices of size ($m, $n) with analytic function f(x) = exp(x)" for
+            (m, n) in ((50, 50), (100, 100), (500, 500), (1000, 1000), (5000, 5000))
+            
+            A = rand(m, n)
+            testMatrix = 0.5 * (A + A')
+            eigMax = eigmax(testMatrix)
+            eigMin = eigmin(testMatrix)
+            condNum = eigMax/eigMin
+            Mp = f(eigMax)
+            mp = f(eigMin)
+            K = ((eigMax - eigMin) * ((sqrt(condNum) - 1)^2) * Mp)/(sqrt(condNum) * mp)
+
+            m = ceil((sqrt(condNum) * log(K/eps)) / 4)
```

Comment:
Perhaps compute `m` and `nv` defaults inside the `slq` function?

Reply:
`m` and `nv` are inputs that user gives, so I'm computing them before hand all the time

Reply:
Yes, but it would be nice to have reasonable default values.